[docs] Create first publishable version of the field doc (#6323)
mui/mui-x: docs/data/date-pickers/fields/fields.md

<p class="description">The fields components let the user select a date / date range with the keyboard.</p>

:::warning
These components are unstable.

> We might do some breaking change on their props to have the best components possible by the time of the stable release.

Reviewer: ```suggestion
They might receive breaking changes on their props to have the best components possible by the time of the stable release.
```

:::

## Introduction

> The fields are React components that lets you enter a date with the keyboard, without any modal or dropdown UI.

Reviewer 2: ```suggestion
The fields are React components that let you enter a date or time with the keyboard, without using any modal or dropdown UI.
They provide refined navigation through arrow keys and support advanced behaviors like localization and validation.
```

### Fields to edit a single element

{{"demo": "SingleDateFieldExamples.js", "defaultCodeOpen": false}}

### Fields to edit a range

All fields to edit a range are available in a single input version and in a multi input version.

{{"demo": "DateRangeFieldExamples.js", "defaultCodeOpen": false}}

### Usage with the pickers

The picker components (`DatePicker`, `DestkopDatePicker`, `MobileDatePicker`, `DateTimePicker`, ...) do not use the field components yet.

New pickers using them will be released in a few months.

## Form props

The standard form attributes (`disabled`, `readOnly`) are supported by all the field components:

{{"demo": "FormProps.js", "defaultCodeOpen": false}}

> ## Advanced

Author: The sections below should probably move to pages not specific to the fields to make them easier to find when the field are stable

But for now I propose to keep the field content inside the scope of the fields, since they are unstable.

Reviewer 3: Yes, and it's the kind of stuf you want people to be aware of when they start using the component

`onChange` will be called if all the sections of the date you modified are filled,
even if some sections of the other date are not filled.

In the demo below, changing the value of a start date section will fire `onChange` even if the end date is empty.
But changing the value of an end date section will not fire `onChange` until the end date is fully completed:

{{"demo": "LifeCycleDateRangeField.js", "defaultCodeOpen": false}}

### Control the selected sections

Use the `selectedSections` and `onSelectedSectionsChange` props to control which sections are currently being selected.

This prop accepts four formats:

1. If a number is provided, the section at this index will be selected.
2. If an object with a `startIndex` and `endIndex` fields are provided, the sections between those two indexes will be selected.
3. If a string of type `MuiDateSectionName` is provided, the first section with that name will be selected.
4. If `null` is provided, no section will be selected

:::warning
You need to make sure the input is focused before imperatively updating the selected sections.
:::

{{"demo": "ControlledSelectedSections.js", "defaultCodeOpen": false }}

:::warning

> For range fields, you will not be able to use the string format since each section is present twice.

Reviewer: I believe we discouraging the use of "will".

```suggestion
For range fields, you can't use the string format since each section is present twice.
```

> We will add new apis in the future to better support this use case.

Reviewer 2: ```suggestion
We'll add new APIs in the future to better support this use case.
```

Reviewer: I think that it would be better without the use of "we" entirely on the page.

```suggestion
New APIs will be added in the future to better support this use case.
```

Reviewer 2: Regarding the "We" rule, It seems there's often a paradox in the writing style.
By not using "we", the chances are that we'll end up in passive voice, which is also discouraged.

In this case, I think the passive voice sounds preferred indeed.

Reviewer: Very true

@samuelsycamore what's the least worse option between using the passive voice and we?

Author: Waiting for Sam feedback for this one
